contracts/rewards/StableLPMaker.sol

```diff
@@ -0,0 +1,172 @@
+// SPDX-License-Identifier: MIT
+// P1 - P3: OK
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "../interfaces/ISwap.sol";
+
+contract StableLpMaker is Ownable {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    ISwap public stableSwap;
+    address public pTri;
+    address private immutable usn;
+    address private immutable usdc;
+    address private immutable usdt;
+    address private immutable tlpToken;
+    address public immutable dao;
+
+    uint256 public polPercent; 
+
+    event LogSetpTri(address oldpTri, address newpTri);
+    event LogSetStableSwap(address oldStableSwap, address newStableSwap);
+
+    event LogWithdrawFees();
+
+    event LogSwapStableTokenToUsn(address stableTokenToConvertToUsn, uint256 stableTokenAmount);
+    event LogAddliquidityToStableSwap(uint256 usnAmount);
+
+    event LogLpTokensSentTopTRI(uint256 tlpAmount);
+    event LogLpTokensSentToDao(uint256 daoAmount);
+    event LogProtocolOwnedLiquidity(uint256 oldpolPercent, uint256 newStableSwap);
+
+    constructor(
+        address _stableSwap,
+        address _pTri,
+        address _usn,
+        address _usdt,
+        address _usdc ,
+        address _tlpToken,
+        address _dao
+    ) public {
+        stableSwap = ISwap(_stableSwap);
+        pTri = _pTri;
+        usn = _usn;
+        usdt = _usdt;
+        usdc = _usdc;
+        tlpToken = _tlpToken;
+        dao = _dao;
+    }
+
+    function setpTri(address _pTri) public onlyOwner {
+        address oldpTri;
+        oldpTri = pTri;
+        pTri = _pTri;
+
+        emit LogSetpTri(oldpTri, _pTri);
+    }
+
+    function setprotocolOwnerLiquidityPercent(uint256 _polPercent) public onlyOwner {
+        require(_polPercent < 50, "POL is too high");
+        uint256 oldpolPercent;
+        oldpolPercent = polPercent;
+        _polPercent = _polPercent;
+
+        emit LogProtocolOwnedLiquidity(oldpolPercent, _polPercent);
+    }
+
```

Comment:
add a function to change `dao` address as well, with `onlyOwner` modifier
also, can you put all the `onlyOwner` functions in one place?

Reply:
done